src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +583,108 @@ Namespace Microsoft.CodeAnalysis.Semantics
+                Dim operandLazy As Lazy(Of IOperation)
+                If operand.Kind = BoundKind.DelegateCreationExpression Then
+                    ' If the child is a BoundDelegateCreationExpression, we don't want to generate a nested IDelegateCreationExpression.
+                    ' So, the operand for the conversion will be the child of the BoundDelegateCreationExpression.
+                    operandLazy = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression)))
+                Else
+                    ' This is either a lambda, or an AddressOf error scenario in which we have a delegate conversion, but it failed to bind correctly.
+                    ' Delegate to standard operation handling for this.
+                    operandLazy = New Lazy(Of IOperation)(Function() Create(operand))
+                End If
+                Return (methodSymbol, operandLazy, isDelegateCreation:=True)
+            Else
+                Dim methodSymbol As MethodSymbol = Nothing
+                Return (methodSymbol, New Lazy(Of IOperation)(Function() Create(operand)), isDelegateCreation:=False)
+            End If
+        End Function
+
+        Private Function CreateBoundDelegateCreationExpressionOperation(boundDelegateCreationExpression As BoundDelegateCreationExpression) As IDelegateCreationExpression
+            Dim syntax As SyntaxNode = boundDelegateCreationExpression.Syntax
+            Dim type As ITypeSymbol = boundDelegateCreationExpression.Type
+            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundDelegateCreationExpression.ConstantValueOpt)
+
+            ' The operand for this is going to be using the same syntax node as this, and since that node can be Explicit, this node cannot be.
+            Dim isImplicit As Boolean = True
+
+            Dim target As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(boundDelegateCreationExpression))
+
+            Return New LazyDelegateCreationExpression(target, _semanticModel, syntax, type, constantValue, isImplicit)
+        End Function
+
+        Private Function CreateBoundDelegateCreationExpressionChildOperation(boundDelegateCreationExpression As BoundDelegateCreationExpression) As IMethodReferenceExpression
+            Dim method As IMethodSymbol = boundDelegateCreationExpression.Method
+            Dim isVirtual As Boolean = method IsNot Nothing AndAlso
+                                               (method.IsAbstract OrElse method.IsOverride OrElse method.IsVirtual) AndAlso
+                                               Not boundDelegateCreationExpression.SuppressVirtualCalls
+            Dim instance As Lazy(Of IOperation) =
+                        New Lazy(Of IOperation)(Function() Create(If(boundDelegateCreationExpression.ReceiverOpt, boundDelegateCreationExpression.MethodGroupOpt?.ReceiverOpt)))
```

Comment:
When is `boundDelegateCreationExpression.ReceiverOpt` null and `boundDelegateCreationExpression.MethodSymbol?.ReceiverOpt` not null?

Reply:
For this, see Binder_Delegates.vb:1049.

---
In reply to: [143331198](https://github.com/dotnet/roslyn/pull/22056#discussion_r143331198) [](ancestors = 143331198)